grafanalib/tests/test_parsing.py

```diff
@@ -0,0 +1,444 @@
+from hypothesis import given, settings, HealthCheck
+import hypothesis.strategies as st
+
+import pytest
+import io
+import json
+import string
+
+import grafanalib.core as G
+import grafanalib._gen as _gen
+
+
+def colors():
+    """Generate arbitrary value from 0 to 255, for RGB color."""
+    return st.integers(min_value=0, max_value=255)
+
+
+def alphas():
+    """Generate alpha values"""
+    return st.floats(min_value=0, max_value=1)
+    # return st.builds(float, st.decimals(min_value=0, max_value=1))
+
+
+def rgbas():
+    """Generate arbitrary, valid RGBA objects."""
+    return st.builds(G.RGBA, r=colors(), g=colors(), b=colors(), a=alphas())
+
+
+def rgbs():
+    """Generate arbitrary, valid RGB objects."""
+    return st.builds(G.RGB, r=colors(), g=colors(), b=colors())
+
+
+def pixels():
+    """Generate arbitrary valid Pixels object"""
+    return st.builds(G.Pixels, num=st.integers(min_value=0))
+
+
+def percents():
+    return st.builds(G.Percent, num=st.integers(min_value=0, max_value=100))
+
+
+def mappings():
+    return st.builds(G.Mapping, name=st.text(), value=st.integers())
+
+
+def unknown():
+    return st.text(string.printable)
+
+
+def grids():
+    return st.builds(G.Grid,
+                     threshold1=unknown(),
+                     threshold1Color=rgbas(),
+                     threshold2=unknown(),
+                     threshold2Color=rgbas(),
+                     leftLogBase=unknown(),
+                     rightLogBase=unknown(),
+                     rightMin=unknown(),
+                     rightMax=unknown(),
+                     leftMin=unknown(),
+                     leftMax=unknown())
+
+
+def legends():
+    return st.builds(G.Legend,
+                     avg=st.booleans(),
+                     current=st.booleans(),
+                     max=st.booleans(),
+                     min=st.booleans(),
+                     show=st.booleans(),
+                     total=st.booleans(),
+                     values=unknown(),
+                     alignAsTable=st.booleans(),
+                     hideEmpty=st.booleans(),
+                     hideZero=st.booleans(),
+                     rightSide=st.booleans(),
+                     sideWidth=unknown(),
+                     sort=unknown(),
+                     sortDesc=unknown())
+
+
+# def targets():
+#     return st.builds(G.Target,
+#                      # target param is usde by st.builds ;-(
```

Comment:
I asked DRMacIver about this and he suggested these options:

* if `target` is a relatively early argument,  you could pass it positionally (I'm not sure we can do this)
* use `composite` 
* use `builds` with `lambda _target, **kwargs: G.Target(target=_target, **kwargs)`

I have no preference between these.

Reply:
Thanks! Will find a way to solve it.

Reply:
Used `composite` seemed more straight forward and very similar to `builds`.